Feature/select (#490)
vtex/styleguide: react/components/Select/README.md

#### A Select lets the user pick one or more options from a list.

### 👍 Dos

> - Mind the order of the options, like putting more probable to be picked on that. In doubt, sort them alphanumerically (from A to Z and from 0 to 9).

Reviewer: ```suggestion
- Mind the order of the options, like putting the more probable one to be selected at the top. In doubt, sort them alphanumerically (from A to Z and from 0 to 9).
```

    label: 'Second Option',
  },
]

;<div>
  <Select
    label="Single option select"
    options={options}
    isMulti={false}
    onChange={values => {
      console.log(`[Select] Selected: ${JSON.stringify(values, null, 2)}`)
    }}
  />
</div>
```

Multi Select

```js
const options = [{
  value: 'first-option',
  label: 'First option',
},
{
  value: 'second-option',
  label: 'Second option',
}];

<div>
  <Select
    label="Label"
    options={options}
    isMulti={true}
    onChange={values => {
      console.log(`[Select] Selected: ${JSON.stringify(values, null, 2)}`)
    }}
  />
</div>
```

With Error

```js
const options = [
  {
    value: 'first-option',
    label: 'First Option',
  },
  {
    value: 'second-option',
    label: 'Second Option',
  },
]

;<div>
  <Select
    label="Label"
    options={options}
    isMulti={true}
    errorMessage="Required!"
    onChange={values => {
      console.log(`[Select] Selected: ${JSON.stringify(values, null, 2)}`)
    }}
  />
</div>
```

Disabled

```js
const options = [
  {
    value: 'first-option',
    label: 'First Option',
  },
  {
    value: 'second-option',
    label: 'Second Option',
  },
]

;<div>
  <Select

>     isDisabled={true}

Reviewer: Manter a mesma API de disabled dos outros componentes do Styleguide:

```suggestion
    disabled={true}
```

    label="Label"
    options={options}
    isMulti={true}
    onChange={values => {
      console.log(`[Select] Selected: ${JSON.stringify(values, null, 2)}`)
    }}
  />
  <div className="mv5">
    <Select
      isDisabled={true}
      label="Label"
      options={options}
      isMulti={true}
      onChange={values => {
        console.log(`[Select] Selected: ${JSON.stringify(values, null, 2)}`)
      }}
      value={[
        {
          value: 'first-option',
          label: 'First Option',
        },
      ]}
    />
  </div>
</div>
```

Loading state

```js
const options = [
  {
    value: 'first-option',
    label: 'First Option',
  },
  {
    value: 'second-option',
    label: 'Second Option',
  },
]

;<div>
  <Select
    isLoading={true}
    label="Label"
    options={options}
    isMulti={true}
    onChange={values => {
      console.log(`[Select] Selected: ${JSON.stringify(values, null, 2)}`)
    }}
  />
</div>

> ```

Reviewer: Senti falta de um exemplo interativo, ou seja, com state e com valores selecionados sendo passados.

Vc pode fazer assim no Styleguidist:

```js
initialState = {
  value: ''
};
<Input
	value={state.value}
	onChange={(e) => {
		setState({ value: e.target.value })
	}
/>
```

Nota que nao tem o `this.`
